dali/c_api/c_api.cc

```diff
@@ -186,16 +186,31 @@ void daliInitialize() {
   std::call_once(init_flag, init);
 }
 
+
 void daliCreatePipeline(daliPipelineHandle *pipe_handle, const char *serialized_pipeline,
                         int length, int max_batch_size, int num_threads, int device_id,
                         int separated_execution, int prefetch_queue_depth,
                         int cpu_prefetch_queue_depth, int gpu_prefetch_queue_depth,
                         int enable_memory_stats) {
-  bool se = separated_execution != 0;
+  daliCreatePipeline2(pipe_handle, serialized_pipeline, length, max_batch_size, num_threads,
+                      device_id, 0, 0, separated_execution == 0 ? 1 : 0, prefetch_queue_depth,
+                      cpu_prefetch_queue_depth, gpu_prefetch_queue_depth, enable_memory_stats);
+}
+
+
+DLL_PUBLIC void
+daliCreatePipeline2(daliPipelineHandle *pipe_handle, const char *serialized_pipeline, int length,
+                    int max_batch_size, int num_threads, int device_id, int pipelined_execution,
+                    int async_execution, int separated_execution, int prefetch_queue_depth,
+                    int cpu_prefetch_queue_depth, int gpu_prefetch_queue_depth,
+                    int enable_memory_stats) {
+  bool se = separated_execution == 0;
+  bool pe = pipelined_execution == 0;
+  bool ae = async_execution == 0;
```

Comment:
OMG, why do you treat 0 as true?
```suggestion
  bool se = separated_execution;
  bool pe = pipelined_execution;
  bool ae = async_execution;
```

Reply:
Done